File: uis/backoffice/types/incidents.ts
export type IncidentCategory =
  | "LOST_PARCEL"
  | "DELAYED_DELIVERY"
  | "WRONG_ADDRESS"
  | "RETURN_REQUEST"
  | "DAMAGE";

export type IncidentStatus = "OPEN" | "CLOSED" | "DISCARDED";
export type TrackFlowCountry = "US" | "ES";

export interface IncidentAnalysisResult {
  total_records: number;
  valid_records: number;
  invalid_records: number;
  category_breakdown: Record<IncidentCategory, number>;
  status_breakdown: Record<IncidentStatus, number>;
  average_satisfaction: number;
  invalid_reasons: Record<string, number>;
  country_breakdown?: Record<TrackFlowCountry, number>;
}

File: services/api/app/models/incidents.py
from pydantic import BaseModel, Field


class IncidentAnalysisResult(BaseModel):
    """JSON contract returned to the TrackFlow backoffice frontend."""

    total_records: int = Field(ge=0)
    valid_records: int = Field(ge=0)
    invalid_records: int = Field(ge=0)

    category_breakdown: dict[str, int]
    status_breakdown: dict[str, int]
    average_satisfaction: float
    invalid_reasons: dict[str, int]
    country_breakdown: dict[str, int]
